components/br-stream/src/metadata/store/etcd.rs

```diff
@@ -20,6 +20,15 @@ use async_trait::async_trait;
 #[derive(Clone)]
 pub struct EtcdStore(Arc<Mutex<etcd_client::Client>>);
 
+impl EtcdStore {
+    pub fn connect<E: AsRef<str>, S: AsRef<[E]>>(endpoints: S) -> Self {
+        // TODO remove block_on
+        let cli =
+            futures::executor::block_on(etcd_client::Client::connect(&endpoints, None)).unwrap();
```

Comment:
Maybe `block_on_external_io` here or create `Endpoint::with_etcd(...)` and reuse the `Runtime` in the `Endpoint`.

Reply:
ok, move connect inside endpoint to reuse runtime.